compiler/daml-extension/BUILD.bazel

```diff
@@ -3,77 +3,65 @@
-    srcs = ["src/extension.ts"],
-    node_modules = "@daml_extension_deps//:node_modules",
-    tsconfig = ":tsconfig.json",
-    deps = [
-        "@daml_extension_deps//@types",
-        "@daml_extension_deps//vscode",
-        "@daml_extension_deps//vscode-languageclient",
-    ],
+sh_test(
+    name = "valid-json",
+    srcs = ["ci-tests.sh"],
+    data = glob([
+        "*.json",
+        "syntaxes/*.json",
+    ]),
 )
 
-# With this rule we get access to extension.js, as
-# the ts_library only has the .d.ts file in the outputs.
-# Could possibly also use filegroup, which allows specifying
-# the output group (es5_source).
-npm_package(
-    name = "out",  # named out, so it goes to same place as before
-    deps = [
-        ":daml_extension_lib",
-    ],
+genrule(
+    name = "node_deps_cache",
+    srcs = ["@daml_extension_deps//:node_modules"],
+    outs = ["node_modules.tar.gz"],
+    cmd = """
+        mkdir node_modules
+        cd node_modules
+        for f in $(locations @daml_extension_deps//:node_modules); do
+            file=$$(basename $$f)
+            dir=$$(dirname $$f | sed 's:^.*/node_modules/::')
+            mkdir -p $$dir
+            cp ../$$f $$dir/$$file
+        done
+        cd ..
+        tar czf node_modules.tar.gz node_modules
+        cp node_modules.tar.gz $@
+    """
 )
 
-pkg_tar(
-    name = "dot-dist",
+genrule(
+    name = "vsix",
     srcs = glob([
         "package.json",
         "syntaxes/*",
         "snippets/*",
         "images/*",
         "*.json",
         "README.md",
+        "yarn.lock",
+        "src/*",
     ]) + [
-        ":out",
-        "src/webview.js",
-        "@daml_extension_deps//vscode-jsonrpc:vscode-jsonrpc",
-        "@daml_extension_deps//vscode-languageclient:vscode-languageclient",
-        "@daml_extension_deps//vscode-languageserver-types:vscode-languageserver-types",
-        "@daml_extension_deps//which:which",
+        ":node_deps_cache",
         "//:VERSION",
     ],
-    extension = "tar.gz",
-    mode = "0755",
-    package_dir = "daml-extension",
-    remap_paths = {
-        "../daml_extension_deps/node_modules": "node_modules",
-    },
-    strip_prefix = "./",
-)
-
-# NOTE(MH): The `pkg_tar` rule puts a `.` at the beginning of every path, which
-# would break assumptions made in the sdk assistant. Hence we need to repack
-# the tarball to get rid of the `.`.
-genrule(
-    name = "dist",
-    srcs = [
-        ":dot-dist",
-        "//:VERSION",
+    tools = [
+        "@daml_extension_deps//vsce/bin:vsce",
+        "@nodejs//:bin/yarn"
     ],
-    outs = ["dist.tar.gz"],
+    outs = ["daml-bundled.vsix"],
     cmd = """
-        tar zxf $(location :dot-dist)
         VERSION=$$(cat $(location //:VERSION))
-        sed -i "s/__VERSION__/$$VERSION/" daml-extension/package.json
-        tar zcf $@ daml-extension
+        cd compiler/daml-extension
+        cp ../../$(location :node_deps_cache) ./node_cache_deps.tar.gz
```

Comment:
I’m not sure why you copy that file at all instead of just using `tar xzf $(location :node_deps_cache)`.

Reply:
I tend to not trust `tar` enough for that. I'll try it.